src/importlinter/domain/helpers.py

```diff
@@ -9,7 +9,7 @@ class MissingImport(Exception):
 
 
 def pop_imports(
-    graph: ImportGraph, imports: Iterable[DirectImport]
+    graph: ImportGraph, imports: Iterable[ImportExpression]
```

Comment:
Let's leave `pop_imports` as it is and have a separate function for converting the ImportExpressions to DirectImports - means the functions are more focused in what they do. (Also, someone may have written a custom contract that uses on `pop_imports`. )

Reply:
As you can see more complex logic is needed. I can make another pop_imports which can accept an ImportExpression. If I reuse this function, the `if not import_details:` will make wildcards unusable

Reply:
> If I reuse this function, the if not import_details: will make wildcards unusable

Ah yes, you might be right that we'll need to tweak `pop_imports` - that check should be moved to the function that converts the ImportExpressions into DirectImports.

Reply:
let me give it a try

Reply:
My currently feeling is, that two functions would make sense: pop_imports with the old logic and pop_import_expressions using import expressions. They share some common code but too little to really have a benefit of code sharing. What do you think?

Reply:
I get your point and I have the same opinion. The problem is this `raise MissingImport(f"Ignored import {import_to_remove} not present in the graph.")`. We need a different check mode for direct imports and import expressions

* direct imports: As soon as one direct import is not found, fail
* import expression: Only fail, if no concrete import matched

E.g. an import expression like `foo*` generates the direct imports `foo1` and `foo2`, but only `foo2` is used in the graph for the import. With your suggestion, using wildcards will basically always result in this error.

One solution I could think of is to convert only one import expression into direct imports at once. Then `pop_imports` can be tuned quite easily to make the exception more manageable. What do you think?

Reply:
I was thinking that `import_expressions_to_imports` would explode wildcarded expressions into direct imports that are actually present in the graph. Correct me if I'm wrong, but would that not solve the issue? Then `pop_imports` would only run into this exception from non-wildcarded expressions, as the wildcarded expressions are guaranteed to pass that check.

Reply:
Got it

Reply:
done, please have a look